design-proposals/vpro-device.md

```diff
@@ -0,0 +1,196 @@
+            ps ->> en: Activate vPRO
+            Note right of en: Provisioning will log the error and update AMT status
+            en ->> rps: vPRO remote configuration
+            activate rps
+            rps ->> en: Done
+            deactivate rps
+            en ->> mps: CIRA connection
+            en ->> ps: Done
+            deactivate en
+        else
+            ps ->> ps: Retry
+        end
+        deactivate ps
+        dm ->> dm: Reconcile the device
+        dm ->> mps: Authorize the device through MPS
+        activate mps
+        mps ->> dm: Done
+        deactivate mps
+        dm ->> inv: Update AMT Status IN_PROGRESS
+        dm ->> inv: Update AMT CurrentState Provisioned
+        deactivate dm
+    else [Device is not eligible]
+        inv ->> ps: Notify
+        activate ps
+        inv ->> dm: Notify
+        activate dm
+        ps ->> ps: Reconcile the device
+        ps ->> en: Provision the device
+        activate en
+        en ->> ps: Done
+        deactivate en
+        deactivate ps
+        dm ->> dm: Reconcile the device
+        deactivate dm
+    end
+    Note right of dm: Check periodically the connection status<br>and use internal timers to track disconnection
+```
+
+**Note1** - MPS requires the creation of a device before accepting CIRA connections which is part of the 2-way auth
+implemented between MPS and AMT;
+
+**Note 2** - Device Management Resource Manager will provide a `staticPassword` profile where the AMT and MEBx
+passwords are set to a well know value. Disabling this option the RM will randomly generate a password for each device.
+
+**Note 3** - Passwords are stored in `Vault` and can be always retrieved either using the Vault APIs or through the
+web-ui.
+
+### MVP Requirements
+
+At the time of writing it is expected to support the following User flows:
+
+- User is able to verify if vPRO/ISM is supported or no;
+- User is able to configure the activation of Edge Node for vPro;
+- User is able to recover the device if something goes wrong during the provisioning of the final OS;
+
+It must be carefully considered the impact on the KPIs as the User will experience worse performance when asking the
+activation of Device Management feature.
+
+However, such flow is considered not mandatory and this penalty might be accepted by the user to have in exchange extra
+manageability features.
+
+## Rationale
+
+Alternative solutions have been carefully considered and will be taken into account in future releases, in particular:
+
+- **Late Binding** where RPC is integrated in the image of the final OS. If this on one side, addresses the need to
+support remote deactivation and day2 reconfiguration, it does not provide a solution to recover the device if something
+goes wrong during the provisioning flow and require the integration with a **Platform Manageability Agent**.
+- **Cloud-Init** provisioning where AMT is activated at the end of the provisioning flow. This solution does not offer
```

Comment:
why would you want this in a future release with all the disadvantages ?

Reply:
Keeping this as alternative. What we will do in 3.2 is the PMA integration